app/models/invitation_manager.rb

```diff
@@ -1,111 +1,59 @@
-        next if student.banned?
-        invitation = Invitation.new(event: event, member: student, role: 'Student')
-        EventInvitationMailer.invite_student(event, student, invitation).deliver_now if invitation.save
-      end
-
-      coaches.each do |coach|
-        next if coach.banned?
-        invitation = Invitation.new(event: event, member: coach, role: 'Coach')
-        EventInvitationMailer.invite_coach(event, coach, invitation).deliver_now if invitation.save
-      end
-    end
+    invite_coaches_to_event(event, chapter) unless event.audience.eql?('Students')
+    invite_students_to_event(event, chapter) unless event.audience.eql?('Coaches')
   end
-
   handle_asynchronously :send_event_emails
 
-  def self.send_monthly_attendance_reminder_emails(monthly)
-    monthly.attendances.map(&:member).each do |member|
-      MeetingInvitationMailer.attendance_reminder(monthly, member)
-    end
-  end
-
-  def self.send_monthly_attendance_reminder_emails(monthly)
-    monthly.attendances.map(&:member).each do |member|
+  def send_monthly_attendance_reminder_emails(monthly)
+    invitees = Member.attending_meeting(monthly)
+    invitees.each do |member|
       MeetingInvitationMailer.attendance_reminder(monthly, member)
     end
   end
+  handle_asynchronously :send_monthly_attendance_reminder_emails
 
-  def self.send_monthly_attendance_reminder_emails(monthly)
-    monthly.attendances.map(&:member).each do |member|
-      MeetingInvitationMailer.attendance_reminder(monthly, member)
-    end
-  end
-
-  def self.send_monthly_attendance_reminder_emails(monthly)
-    monthly.attendances.map(&:member).each do |member|
-      MeetingInvitationMailer.attendance_reminder(monthly, member)
-    end
-  end
-
-  def self.send_course_emails(course)
-    course.chapter.groups.students.map(&:members).flatten.uniq.each do |student|
+  def send_course_emails(course)
+    students = Member.not_banned.joins(:groups).merge(course.chapter.groups.students)
+    students.each do |student|
       invitation = CourseInvitation.new(course: course, member: student)
       invitation.send(:email) if invitation.save
     end
   end
+  handle_asynchronously :send_course_emails
 
-  def self.send_workshop_attendance_reminders(workshop)
+  def send_workshop_attendance_reminders(workshop)
     workshop.attendances.where(reminded_at: nil).each do |invitation|
       WorkshopInvitationMailer.attending_reminder(workshop, invitation.member, invitation).deliver_now
       invitation.update_attribute(:reminded_at, Time.zone.now)
     end
   end
+  handle_asynchronously :send_workshop_attendance_reminders
 
-  def self.send_workshop_waiting_list_reminders(workshop)
-    # Only send out reminders to people where reminded_at is nil, ie. falsey.
-    workshop.waiting_list.reject(&:reminded_at).each do |invitation|
+  def send_workshop_waiting_list_reminders(workshop)
+    workshop.invitations.where(reminded_at: nil).joins(:waiting_list).each do |invitation|
```

Comment:
Is this an opportunity to use a scope?  "unsent_invitations" or something like that?